elec/src/LM5164_regulator.ato

```diff
@@ -0,0 +1,230 @@
+from "generics/resistors.ato" import Resistor
+from "generics/diodes.ato" import SchottkyDiode
+
+component LM5164DDAR:
+    # component LM5164DDAR
+    """ Synchronous step-down buck regulator 6 to 100 Vin, 1A DC load current, 10.5uA Iq """
+    footprint = "SO-8_L4.9-W3.9-P1.27-LS6.0-BL-EP"
+    lcsc_id = "C477928"
+    mpn = "C477928"
+    # pins
+    signal GND ~ pin 1
+    signal VIN ~ pin 2
+    signal EN_slash_UVLO ~ pin 3
+    signal RON ~ pin 4
+    signal FB ~ pin 5
+    signal PGOOD ~ pin 6
+    signal BST ~ pin 7
+    signal SW ~ pin 8
+    signal EP ~ pin 9
+    EP ~ GND # EP is floating, but for heatsinking we connect to the GND plane
+
+
+module LM5164Buck from AdjustableRegulator:
+    """
+    This follows the configuration here:
+    https://www.ti.com/lit/ds/symlink/lm5164.pdf
+
+    Usage:
+    - set v_in and v_out
+    - [optinal] set the additional "top-level" params below
+    - replace (->) the inductor and diode with the actual components
+    - set the inductor.value
+    - set the diode.voltage and diode.current
+    - set c_out value
+    - replace c_in_bulk and c_in_hf with actual components,
+        which should be 2x the voltage of v_in
+
+    """
+
+    # Additional top-level params to configure
+    output_current = 0 to 1A
+    output_ripple = 50mV
+    input_ripple = 400mV
+    switching_frequency = 500kHz ± 3%
+
+    assert v_out < v_in <= 100V
+    assert output_current <= 1A
+    assert switching_frequency within 200kHz to 1MHz
+
+    ic = new LM5164DDAR
+
+    # Power Input
+    power_in.vcc ~ ic.VIN
+    power_in.gnd ~ ic.GND
+
+    # Output Voltage Set-Point
+    # The remainder of this is handeled in our super class
+    power_out ~ feedback_div.power
+    feedback_div.output.io ~ ic.FB
+    v_ref = 1.2V
+    # Force bottom value so we get expected resistance for ripple generation
+    feedback_div.r_bottom.value = 51kΩ +/- 1%
+    i_q = 20uA to 100uA 
+    # TI recommends:
+    assert feedback_div.r_top.value within 100 kΩ to 1 MΩ
+
+    # Feedback Ripple Generation method: Type 3 
+    # This method is chosen since it does not rely on output capacitor ESR for stability
+    Ra = new Resistor
+    Ca = new Capacitor
+    Cb = new Capacitor
+    Ra.p1 ~ ic.SW
+    Ca.p1 ~ power_out.vcc
+    Ra.p2 ~ Ca.p2
+    Cb.p1 ~ Ca.p2
+    Cb.p2 ~ ic.FB
+    #assert Cb.ceramic = "C0G"
+    Ra.package = "0402"
+    Ca.package = "0402"
+    Cb.package = "0402"
+
+    # Check values against datasheet equations:
+    tTR_setting = 75us # Desired load transient response settling time
+    # Equation 24:
+    assert Ca.value >= 10 / (switching_frequency * 1/(1/feedback_div.r_bottom.value + 1/feedback_div.r_top.value))
+    # Equation 25:
+    Ton = (r_switching_frequency.value / 1kΩ) / (v_in/1V * 2.5) * 1us
+    Ton_min = 70 / (48 * 2.5) * 1us
+    Ton_max = 70 / (68 * 2.5) * 1us
+    assert Ra.value * Ca.value <= (v_in - v_out) * Ton_max / 20mV # Using Ton here breaks the build?!
```

Comment:
@mawildoer I was having trouble with this line, could you look into it?

Reply:
Is it building on your end?

Reply:
`ato build -b lm5164_buck_demo` is working on the latest commit
default isn't working, but I think it's broken in main too so I didn't try too hard there